### PR TITLE
Feature vpc

### PR DESCRIPTION
**Pull request by sunilsunderrajdev**
VPC Task

### CODE REVIEW

File: .github/workflows/main..yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Terraform Apply
+        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+        run: terraform apply -auto-approve
+#      - name: Terraform Plan Destroy
+#        id: plandestroy
+#        run: terraform plan -destroy
+#        continue-on-error: true        
+#      - name: Terraform Destroy
+#        if: github.ref == 'refs/heads/feature_vpc' && github.event_name == 'push'
+#        run: terraform apply -destroy
```
Comment (geoffrey1330) [inline .github/workflows/main..yml]: could you please add a newline at the end of file.

File: ec2.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        Name: "YDM EC2 in public subnet 1 - Bastion host"
+    }
+
+    depends_on = [aws_key_pair.EC2_key_pair_nonaws]
+}
+
+# KeyPair
+resource "aws_key_pair" "EC2_key_pair_nonaws" {
+    key_name = "EC2_key_pair_nonaws"
+    public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDGYsZQumFVfIvHlIS/UppkAhfu5N9vZ0kMFcZEviH3TXEgXZ5pChUPTWjcEhViq5MJhJS4UfvBNluPd3K8UGnnduaeRoZ9yURYp2Sy39EZvonVbj0zoi3SxpmKHif9u4f/Az8WxalBKRn8cz1+ALyeiJwZFSeVqfJ9Oz5KcwWUHibDP+QBd8zJcGVJjaosOuRgiCDK68lwDJcpzx0Rqxmj1HlyL8cJCshoPqnX8Tc8D1pDSPVGnbkbrXBcEAgBG+sFoVyqXxsn2CZ3FEwoBZegdsqOoucen8x3FwSKA1HxL0EbocklovO8xedBu0/GB57je265+EYrxKGdm1kwWSTT shikh@SSRDROID"
```
Comment (geoffrey1330) [inline ec2.tf]: could you please remove the hardcoded key and associate a keypair to this ec2 instance.

File: ec2.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+resource "aws_key_pair" "EC2_key_pair_bastion" {
+    key_name = "EC2_key_pair_bastion"
+    public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCmkm4M7pgDMhb8z7GKFl2SDBwoe8AiMaJ19H1y+L2s7hNCqhnfkoykv9vqBBdHkCHOIfHd7Zz0B3qIsI4RrOSanXDWsoV4WBOfIXvIAbVOk3BIA4bDQoh3OfPawIrvYceNPun05F106vqho69iBBe6PmoOT9YTTTKYy5cUhAYefGN/npB4lrOgl59s0GUHmhBUvt30tAkXbH/Rn4WjH6Ulmzsjn0xUZDR9r6PSpU+s2vK18h6z76ZJap1BpmLtdSE85SrZdRAeaySR695O++Dx3sLD20EbzpDnrnj6kVh5brbMRXaI9a9pdqUFOqVRyOiajYDAAVSI/hDytdKROw3peRhtXYNT1bXXQwizq/GG8vr5nFEEsUjMNxI7UX9RPAHzbXu1ssKhVxskXQ+Dt+MNCFOa+7EQLt96QB2NRWeLicyUx+7k06z8dXHgPzhzl0ToyISASXIXC0jr6VYofpSgrJdtmDjCHJXsDSTw3lCDUS+zDuv9jtcK20KXc5oQgE0= ec2-user@ip-10-0-1-70.ec2.internal"
+}
```
Comment (geoffrey1330) [inline ec2.tf]: could you please add a newline at the end of file.

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,5 @@
+resource "aws_ssm_parameter" "foo" {
+    name = "foo"
+    type = "String"
+    value = "bar"
+}
```
Comment (geoffrey1330) [inline main.tf]: could you please add a newline at the end of file.

File: ec2.tf [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    key_name = "EC2_key_pair_nonaws"
+    public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDGYsZQumFVfIvHlIS/UppkAhfu5N9vZ0kMFcZEviH3TXEgXZ5pChUPTWjcEhViq5MJhJS4UfvBNluPd3K8UGnnduaeRoZ9yURYp2Sy39EZvonVbj0zoi3SxpmKHif9u4f/Az8WxalBKRn8cz1+ALyeiJwZFSeVqfJ9Oz5KcwWUHibDP+QBd8zJcGVJjaosOuRgiCDK68lwDJcpzx0Rqxmj1HlyL8cJCshoPqnX8Tc8D1pDSPVGnbkbrXBcEAgBG+sFoVyqXxsn2CZ3FEwoBZegdsqOoucen8x3FwSKA1HxL0EbocklovO8xedBu0/GB57je265+EYrxKGdm1kwWSTT shikh@SSRDROID"
+}
+
+resource "aws_key_pair" "EC2_key_pair_bastion" {
+    key_name = "EC2_key_pair_bastion"
+    public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCmkm4M7pgDMhb8z7GKFl2SDBwoe8AiMaJ19H1y+L2s7hNCqhnfkoykv9vqBBdHkCHOIfHd7Zz0B3qIsI4RrOSanXDWsoV4WBOfIXvIAbVOk3BIA4bDQoh3OfPawIrvYceNPun05F106vqho69iBBe6PmoOT9YTTTKYy5cUhAYefGN/npB4lrOgl59s0GUHmhBUvt30tAkXbH/Rn4WjH6Ulmzsjn0xUZDR9r6PSpU+s2vK18h6z76ZJap1BpmLtdSE85SrZdRAeaySR695O++Dx3sLD20EbzpDnrnj6kVh5brbMRXaI9a9pdqUFOqVRyOiajYDAAVSI/hDytdKROw3peRhtXYNT1bXXQwizq/GG8vr5nFEEsUjMNxI7UX9RPAHzbXu1ssKhVxskXQ+Dt+MNCFOa+7EQLt96QB2NRWeLicyUx+7k06z8dXHgPzhzl0ToyISASXIXC0jr6VYofpSgrJdtmDjCHJXsDSTw3lCDUS+zDuv9jtcK20KXc5oQgE0= ec2-user@ip-10-0-1-70.ec2.internal"
```
Comment (geoffrey1330) [inline ec2.tf]: could you please remove the hardcoded key and associate a keypair to this ec2 instance.

File: network.tf [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+ availability_zone_id = element(["use1-az1"], count.index)
+ map_public_ip_on_launch = true
+ 
+ tags = {
+   Name = "Public Subnet ${count.index + 1}"
+ }
+}
+
+resource "aws_subnet" "private_subnets" {
+ count      = length(["10.0.3.0/24","10.0.4.0/24"])
```
Comment (geoffrey1330) [inline network.tf]: declare this cidr in a var or terraform locals and then make i reference to it here.

File: network.tf [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  depends_on    = [aws_internet_gateway.igw]
+  tags = {
+    Name        = "Public nat"
+  }
+}
+
+resource "aws_subnet" "public_subnets" {
+ count      = length(["10.0.1.0/24","10.0.2.0/24"])
+ vpc_id     = aws_vpc.vpc.id
+ cidr_block = element(["10.0.1.0/24","10.0.2.0/24"], count.index)
```
Comment (geoffrey1330) [inline network.tf]: declare this cidr in a var or terraform locals and then make i reference to it here

File: network.tf [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  ingress {
+      from_port = 22
+      to_port = 22
+      protocol = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags = {
+      Name = "ssh-allowed"
+  }
+}
```
Comment (geoffrey1330) [inline network.tf]: could you please add a newline at the end of file.

File: outputs.tf [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#output "public_key" {
+#  value       = join("", tls_private_key.EC2_private_key.*.public_key_openssh)
+#  description = "Content of the generated public key"
+#}
+
+#output "private_key" {
+#  value       = tls_private_key.EC2_private_key.private_key_pem
+#  description = "Content of the generated private key"
+#  sensitive   = true
+#}
```
Comment (geoffrey1330) [inline outputs.tf]: could you please add a newline at the end of file.

File: providers.tf [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+        workspaces {
+            name = "Github-AWS-TF_API"
+        }
+    }
+}
+
+provider "aws" {
+    region = "us-east-1"
+}
```
Comment (geoffrey1330) [inline providers.tf]: could you please add a newline at the end of file.

File: variables.tf [hunk 11]
```diff
@@ -0,0 +1,9 @@
+variable "region" {
+    type    = string
+    default = "us-east-1"
+}
+
+variable "vpc_cidr" {
+    type    = string
+    default = "10.0.0.0/16"
+}
```
Comment (geoffrey1330) [inline variables.tf]: could you please add a newline at the end of file.

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

**Comment by sunilsunderrajdev** [comment]
Changes requested

**Pull request closed without merging**